src/test/java/org/pentaho/metastore/test/MetaStoreFactoryTest.java

```diff
@@ -560,15 +560,20 @@ private List<DimensionAttribute> generateAttributes() {
   }
 
   @Test
-  public void testGetterNotExists() {
+  public void testElementsWithFieldsInParent() throws MetaStoreException {
     IMetaStore metaStore = new MemoryMetaStore();
-    MetaStoreFactory<MissingGetterElement> factory =
-      new MetaStoreFactory<>( MissingGetterElement.class, metaStore, "custom" );
-    MissingGetterElement element = new MissingGetterElement( ANOTHER, ATTR );
-    try {
-      factory.saveElement( element );
-    } catch ( MetaStoreException e ) {
-      assertNotNull( e );
-    }
+    MetaStoreFactory<InheritedElement> factory =
+      new MetaStoreFactory<>( InheritedElement.class, metaStore, "custom" );
+    InheritedElement element = new InheritedElement( ANOTHER, ATTR );
+    element.setName( "name" );
+
+    assertEquals( 0, factory.getElements().size() );
+
+    factory.saveElement( element );
+
+    List<InheritedElement> elements = factory.getElements();
```

Comment:
It would be nice to also test that fields of the subclass are serialized as expected.

Also, another test is to make sure that duplicate/overridden private fields also work. Would better create another field in the base class for this test, perhaps.

Reply:
Addressed